tpu-client/src/nonblocking/tpu_client.rs

```diff
@@ -528,10 +530,7 @@ where
             .await
             {
                 Ok(service) => return Ok(service),
-                Err(_) => {
-                    warn!("Failed to create TpuLeaderService. Will retry in 1 second");
-                    sleep(Duration::from_secs(1)).await;
-                }
+                Err(_) => sleep(Duration::from_secs(1)).await,
```

Comment:
How hard is it to check for the specific error that indicates the leader schedule is not ready just yet?

I think it is better to avoid hiding other kinds of errors, if all we want is to retry for a specific error that we know is retriable and is part of the expected execution.